File: YABA-New/YABA/app/src/main/java/dev/subfly/yaba/core/state/home/HomeEvent.kt
package dev.subfly.yaba.core.state.home

import dev.subfly.yaba.core.model.ui.BookmarkUiModel
import dev.subfly.yaba.core.model.ui.FolderUiModel
import dev.subfly.yaba.core.model.ui.TagUiModel
import dev.subfly.yaba.core.model.utils.BookmarkAppearance
import dev.subfly.yaba.core.model.utils.CardImageSizing
import dev.subfly.yaba.core.model.utils.SortOrderType
import dev.subfly.yaba.core.model.utils.SortType

sealed class HomeEvent {
    // Initialization
    data object OnInit : HomeEvent()

    // Preference changes
    data class OnChangeBookmarkAppearance(val appearance: BookmarkAppearance) : HomeEvent()
    data class OnChangeCardImageSizing(val sizing: CardImageSizing) : HomeEvent()
    data class OnChangeCollectionSorting(val sortType: SortType) : HomeEvent()
    data class OnChangeSortOrder(val sortOrder: SortOrderType) : HomeEvent()

    // Folder operations
    data class OnToggleFolderExpanded(val folderId: String) : HomeEvent()
    data class OnDeleteFolder(val folder: FolderUiModel) : HomeEvent()
    data class OnMoveFolder(
        val folder: FolderUiModel,
        val targetParent: FolderUiModel?,
    ) : HomeEvent()

    // Tag operations
    data class OnDeleteTag(val tag: TagUiModel) : HomeEvent()

    // Bookmark operations
    data class OnDeleteBookmark(val bookmark: BookmarkUiModel) : HomeEvent()
    data class OnMoveBookmarkToFolder(
        val bookmark: BookmarkUiModel,
        val targetFolder: FolderUiModel,
    ) : HomeEvent()

    data class OnMoveBookmarkToTag(
        val bookmark: BookmarkUiModel,
        val targetTag: TagUiModel,
    ) : HomeEvent()
}

File: Darwin/YABACore/State/Home/HomeEvent.swift
//
//  HomeEvent.swift
//  YABACore
//
//  Parity with Compose `HomeEvent` — IDs only; list rows come from `@Query`.
//

import Foundation

public enum HomeEvent: Sendable {
    case onInit

    // Preferences (host may mirror to `UserDefaults` / `@AppStorage`)
    case onChangeBookmarkAppearance(BookmarkAppearance)
    case onChangeCardImageSizing(CardImageSizing)
    case onChangeCollectionSorting(SortType)
    case onChangeSortOrder(SortOrderType)

    case onToggleFolderExpanded(folderId: String)
    case onDeleteFolder(folderId: String)
    case onMoveFolder(folderId: String, newParentFolderId: String?)

    case onDeleteTag(tagId: String)

    case onDeleteBookmark(bookmarkId: String)
    case onMoveBookmarkToFolder(bookmarkId: String, targetFolderId: String)
    /// Same as Compose `OnMoveBookmarkToTag` — adds `targetTag` association.
    case onMoveBookmarkToTag(bookmarkId: String, targetTagId: String)
}
